### PR TITLE
Rewrite an rbac test with gen_user fixture

### PR DESCRIPTION
**Pull request by mdellweg**
Demonstration for: https://github.com/pulp/pulp-smash/pull/1279

### CODE REVIEW

File: pulp_container/tests/functional/api/test_rbac_sync_repositories.py [hunk 1]
```diff
@@ -1,98 +1,81 @@
...
+    with user2, pytest.raises(ApiException):
+        container_repository_api.delete(repository.pulp_href)
+    with user3, pytest.raises(ApiException):
+        container_repository_api.delete(repository.pulp_href)
+    with user1:
+        response = container_repository_api.delete(repository.pulp_href)
         monitor_task(response.task)
-        with self.assertRaises(ApiException):
-            self.user1["repository_api"].read(self.repository.pulp_href)
+        with pytest.raises(ApiException):
```
Comment (bmbouter) [inline pulp_container/tests/functional/api/test_rbac_sync_repositories.py:80]: nice I didn't realize this was part of pytest.

Comment (mdellweg) [reply pulp_container/tests/functional/api/test_rbac_sync_repositories.py:80]: Well they sure needed to provide something like this.

**Comment by lubosmj** [comment]
I wonder if we could extract particular tests into separate functions. Right now, we have all scenarios included within a single function. We can create fixtures for users and pass them to those separate functions. Then, we would have something like this (`test_rbac_remotes.py`):
```
def test_create_remote(users, container_api):
    ...

def test_read_remote(users, container_api):
    ...

def test_update_remote(users, container_api):
    ...
```

Is it worth the time?

**Comment by mdellweg** [comment]
> I wonder if we could extract particular tests into separate functions. Right now, we have all scenarios included within a single function. We can create fixtures for users and pass them to those separate functions. Then, we would have something like this (`test_rbac_remotes.py`):
> 
> ```
> def test_create_remote(users, container_api):
>     ...
> 
> def test_read_remote(users, container_api):
>     ...
> 
> def test_update_remote(users, container_api):
>     ...
> ```
> 
> Is it worth the time?

This is not even about the users. This is about the lifecycle of the remote under test. You need it created to update and later delete it. The fact that we don't need to recreate the users for every single test is just a nice plus.

**Pull request merged**